src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs

```diff
@@ -274,6 +274,37 @@ public static string GetTempPath(string workingDir)
             }
         }
 
+        private static string FindDotNetExecutable()
+        {
+            var expectedPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH");
+            if (!string.IsNullOrEmpty(expectedPath))
+            {
+                return expectedPath;
+            }
+
+#if NET
+            expectedPath = System.Environment.ProcessPath;
+#else
+            expectedPath = Process.GetCurrentProcess().MainModule.FileName;
+#endif
+
+            if (!Path.GetFileNameWithoutExtension(expectedPath).Equals("dotnet"))
+            {
+                // We were probably running from Visual Studio or Build Tools and found MSBuild instead of dotnet. Use the PATH...
+                var paths = Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator);
+                foreach (string path in paths)
+                {
+                    var dotnetPath = Path.Combine(path, "dotnet.exe");
+                    if (File.Exists(dotnetPath))
+                    {
+                        return dotnetPath;
+                    }
+                }
+            }
+
+            return File.Exists(expectedPath) && Path.GetFileNameWithoutExtension(expectedPath).Equals("dotnet") ? expectedPath : "dotnet";
```

Comment:
I think there's a redundant clause here given the previous if statement.

```suggestion
            return File.Exists(expectedPath) ? expectedPath : "dotnet";
```

Reply:
That clause is for if you are running from VS and find MSBuild rather than dotnet, and dotnet is not on your PATH for some reason. If you have a custom assembly resolve hook, I could theoretically imagine this being important...though I'll admit that I'm thinking < 1% of cases.

Reply:
> if you are running from VS and find MSBuild rather than dotnet, and dotnet is not on your PATH

Why not simply do this?

```cs
if (Path.GetFileNameWithoutExtension(expectedPath) != "dotnet" || !File.Exists(expectedPath))
{
    // Find dotnet on PATH
    foreach { ... }

    // Fallback
    return "dotnet";
}

return expectedPath;
```

Reply:
On that note, the File.Exists check is unnecessary, too, since the current process's path should exist. I refactored this a bit, and it's cleaner now in my opinion. Thanks for the suggestions 🙂